toolchain/lowering/testdata/function/call/var_param.carbon

```diff
@@ -22,7 +22,8 @@ fn Main() {
 // CHECK:STDOUT:   %var = alloca i32, align 4
 // CHECK:STDOUT:   store i32 0, ptr %var, align 4
 // CHECK:STDOUT:   %1 = load i32, ptr %var, align 4
-// CHECK:STDOUT:   call void @DoNothing(i32 %1)
+// CHECK:STDOUT:   %2 = load i32, ptr %var, align 4
```

Comment:
Any idea why this results in a redundant / dead load here?

(not blocking regardless, happy to fix forward)

Reply:
We weren't properly updating the call argument list when argument conversions were required. Fixed.